Vignettes/using_cchsflow.Rmd

```diff
@@ -0,0 +1,286 @@
+---
+title: "Using cchsflow"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{1 - using cchsflow}     
+  %\VignetteEngine{knitr::rmarkdown} 
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Getting Started
+
+### Loading the necessary packages
+
+The `rec_with_table()` and `set_data_labels()` functions are part of cchsflow. `rec_with_table()` is used to transform CCHS variables to a common name. `set_data_labels()` reapplies labels that are lost during the combination process after transforming each CCHS dataset. `variables.csv` & `variable_details.csv` are also saved in the cchsflow package as `variables` & `variable_details`. This means you will need to install & load the page onto your environment. Install & load the package onto your environment.
+
+```{r eval= FALSE}
+install.packages("devtools")
+library(devtools)
+install_github("Big-Life-Lab/cchsflow")
+```
+
+```{r results= 'hide', message = FALSE, warning=FALSE}
+library(cchsflow)
+```
+
+We use the `bind_rows` function in the [dplyr](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8) package to combine datasets.
+
+```{r results = 'hide', message = FALSE, warning =FALSE}
+library(dplyr)
+```
+
+For this vignette, we use the `kable` function in the [knitr](https://yihui.name/knitr/) package to illustrate transformation.
+```{r results = 'hide', message = FALSE, warning =FALSE}
+library(knitr)
+library(kableExtra)
+```
+
+## Steps
+
+### Step 1. Load the CCHS datasets
+
+Load the CCHS datasets as well as the two variable worksheets. For illustration, we will create mock CCHS datasets for the years 2001 and 2013. 
+
+```{r}
+cchs_mock2001 <- data.frame(DHHA_SEX = c(2, 1, 1, 6), DHHAGAGE = c(3, 4, 6, 6), FVCADTOT = c(12, 4, 20, 9))
+cchs_mock2013 <- data.frame(DHH_SEX = c(1, 2, 1), DHHGAGE = c(2, 1, 1), FVCDTOT = c(25, 15, 6))
+```
+
+Did you notice that the names for the variables are slightly different in the two mock databases? That isn't a mistake: in the 2001 CCHS the variable for sex is `DHHA_SEX` and in 2013 CCHS the variable is `DHH_SEX`. 
+
+Don't worry, `cchsflow` is here to help! `variable_details.csv` contains the rules to harmonize those two variables into a common variable name. In the CCHS, the categories for `sex` are consistient: 1 = males, 2 = females. `variable_details.csv` provides instructions for how to harmonizecategory values or labels, if they change across survey cycles. Variable labes are discussed in later vignettes.
+
+```{r, echo=FALSE, message=FALSE, warning=FALSE}
+library(DT)
+datatable(variables, options = list(pageLength = 5))
```

Comment:
didnt u switch to kable?

Reply:
Kable doesn't limit the number of rows on a page. For this example only I used `datatable()`